include/grpcpp/impl/codegen/sync.h

```diff
@@ -56,13 +54,7 @@ class Mutex {
   const gpr_mu* get() const { return &mu_; }
 
  private:
-  union {
-    gpr_mu mu_;
-    std::mutex do_not_use_sth_;
-#ifdef GPR_HAS_PTHREAD_H
-    pthread_mutex_t do_not_use_pth_;
-#endif
-  };
+  gpr_mu mu_;
 };
```

Comment:
This one is a special case. We have to keep this one. We need a union that is as big as the bigger of gpr_mu, std::mutex, and pthread_mutex_t in order to be able to run multi-platform. This has caused problems for us in the past.

Reply:
Can you elaborate this a bit more? It isn't clear to me why it's required. Does gRPC support link-time multi-platform such as building codegen with pthread and linking it with something with pthread-alternative (like absl::mutex)?

Reply:
Chat with @soheilhy about this topic.

Reply:
I did actually and it seems that there was ODR violation and it required this hack to work it around. But we're not sure it's still required. I'll try to remove this later.

Reply:
Thank you. It'd be great if we can make sure this never happens and then get rid of this hack!